pkg/sql/catalog/resolver/resolver_bench_test.go

```diff
@@ -113,3 +114,88 @@ func BenchmarkResolveExistingObject(b *testing.B) {
 		})
 	}
 }
+
+// BenchmarkResolveFunction exercises resolver code to ensure that
+// we do not regress in this important hot path.
+func BenchmarkResolveFunction(b *testing.B) {
+	defer leaktest.AfterTest(b)()
+	defer log.Scope(b).Close(b)
+	for _, tc := range []struct {
+		setup      []string
+		name       tree.UnresolvedName
+		flags      tree.ObjectLookupFlags
+		searchPath string
+	}{
+		{
+			setup: []string{
+				"CREATE FUNCTION foo() RETURNS int IMMUTABLE LANGUAGE SQL AS $$ SELECT 1 $$",
+			},
+			name:  tree.MakeUnresolvedName("foo"),
+			flags: tree.ObjectLookupFlagsWithRequired(),
+		},
+		{
+			setup: []string{
+				"CREATE SCHEMA sc",
+				"CREATE FUNCTION sc.foo() RETURNS int IMMUTABLE LANGUAGE SQL AS $$ SELECT 1 $$",
+			},
+			name:  tree.MakeUnresolvedName("sc", "foo"),
+			flags: tree.ObjectLookupFlagsWithRequired(),
+		},
+		{
+			setup: []string{
+				"CREATE SCHEMA sc",
+				"CREATE FUNCTION sc.foo() RETURNS int IMMUTABLE LANGUAGE SQL AS $$ SELECT 1 $$",
+			},
+			name:       tree.MakeUnresolvedName("foo"),
+			flags:      tree.ObjectLookupFlagsWithRequired(),
+			searchPath: "public,$user,sc",
+		},
+	} {
+		b.Run(strings.Join(tc.setup, ";")+tc.name.String(), func(b *testing.B) {
```

Comment:
the test name will probably be fun. probably manually give them some more readable names.